### ISSUE TITLE
metrics  in the Classification on imbalanced data tutorial 

### ISSUE THREAD

**Issue by louridas** [tensorflow/tensorflow#45615]
Thank you for submitting a TensorFlow documentation issue. Per our GitHub
policy, we only address code/doc bugs, performance issues, feature requests, and
build/installation issues on GitHub.

The TensorFlow docs are open source! To get involved, read the documentation
contributor guide: https://www.tensorflow.org/community/contribute/docs

## URL(s) with the issue:

https://www.tensorflow.org/tutorials/structured_data/imbalanced_data

## Description of issue (what needs changing):

The example uses the AUC-ROC metric, both for evaluation and as the criterion for early stopping. However, for imbalanced datasets the Precision-Recall curve may be a better choice. In fact, this is the metric that is recommended in the Kaggle dataset (https://www.kaggle.com/mlg-ulb/creditcardfraud), see also:

https://www.biostat.wisc.edu/~page/rocpr.pdf

### Clear description

Instead of using the ROC-AUC, the PRC Area Under the Curve should be used instead. In particular, 

* The `METRICS` array should include:

    `keras.metrics.AUC(name='prc', curve='PR'), # precision-recall curve`

* `early_stopping` should be defined as:

    ```
    early_stopping = tf.keras.callbacks.EarlyStopping(
        monitor='val_prc', 
        verbose=1,
        patience=10,
        mode='max',
        restore_best_weights=True)
   ```

* The first line of `plot_metrics(history)` should become:

    ```
    metrics = ['loss', 'prc', 'precision', 'recall']
    ```

* Instead of having a `plot_auc()` we should have:

    ```
    def plot_prc(name, labels, predictions, **kwargs):
        precision, recall, _ = sklearn.metrics.precision_recall_curve(labels, predictions)

        plt.plot(precision, recall, label=name, linewidth=2, **kwargs)
        plt.xlabel('Recall')
        plt.ylabel('Precision')
        plt.grid(True)
        ax = plt.gca()
        ax.set_aspect('equal')
    ```

### Submit a pull request?

No, because I have also made some other changes, the most important of which is that oversampling does not seem to give very good results, but a combination of SMOTE oversampling and undersampling does give better results. This is easy to do with imbalanced-learn (I might submit another issue). Also, I am not sure why the adjusted output bias is used when we reweight the classes, as the effect is the same as with oversampling, and we zero out the last bias in oversampling.

**Comment by rmothukuru** [comment]
@louridas,
Thanks for the suggestion.  Can you please click on "**`Run in Google Colab`**" option of the [Tutorial](https://www.tensorflow.org/tutorials/structured_data/imbalanced_data), do the respective changes and share the **`Github Gist`** (File -> Save a Copy as Github Gist). 

Also, if possible, please highlight how this change is better over he existing approach.

Thanks!

**Comment by louridas** [comment]
Hello, I did not have time to do the respective changes as of yet, but please do not close the issue, I will do it.

**Comment by louridas** [comment]
Same, will get to it as soon as I find some time.

**Comment by samaritanhu** [comment]
Hi @rmothukuru, I am very interested in this issue. Can you assign this issue to me? Thanks!

**Comment by samaritanhu** [comment]
Hi @rmothukuru, I have already made the changes in my own gist. I don't use gist very often... I don't know how to commit my changes through gist. 
My former experience is :
pull request my own branch, commit to my branch, pull request to the main branch.

But for this doc, I don't know where it is located. Need help here. Thanks!

**Comment by rmothukuru** [comment]
@samaritanhu,
Thank you for your contribution. 

The Tutorial is located in the path, https://github.com/tensorflow/docs/blob/master/site/en/tutorials/structured_data/imbalanced_data.ipynb.

Please find the below links for the doc changes:

contributor guide: https://www.tensorflow.org/community/contribute/docs,
docs API guide: https://www.tensorflow.org/community/contribute/docs_ref and the
docs style guide: https://www.tensorflow.org/community/contribute/docs_style

**Comment by samaritanhu** [comment]
Thanks, @rmothukuru! I have already pull request into the tensorflow/doc here: https://github.com/tensorflow/docs/pull/1847.

**Comment by iteachmachines** [comment]
@rmothukuru  I would love to work on this issue.Can you assign this issue to me?

**Comment by MarkDaoust** [comment]
This was fixed by tensorflow/docs#1847
